interfaces/input_parameters/ring_section.py

```diff
@@ -177,98 +165,134 @@ def __init__(self, section_length, alpha_0,
+        if not isinstance(alpha_0, ring_programs.momentum_compaction):
+            alpha_0 = ring_programs.momentum_compaction(alpha_0, order=0)
         else:
             if alpha_0.order != 0:
                 raise excpt.InputError(
                     "The order of the datatype passed as keyword " +
                     "argument alpha_%s do not match" % (0))
 
-        self._check_and_set_momentum_compaction(alpha_0, 0)
+        self._check_and_set_alpha_and_orbit(alpha_0, 0)
 
         # Treating non-linear momentum compaction factor if declared
         # Listing all the declared alpha first
         alpha_n = {1: alpha_1, 2: alpha_2}
 
         if alpha_1 is not None:
-            self.alpha_order = 1
+            alpha_order_max = 1
         if alpha_2 is not None:
-            self.alpha_order = 2
+            alpha_order_max = 2
 
         for argument in kwargs:
             if 'alpha' in argument:
                 try:
                     order = int(argument.split('_')[-1])
-                    self.alpha_order = np.max([self.alpha_order, order])
+                    alpha_order_max = np.max([alpha_order_max, order])
                     alpha_n[order] = kwargs[argument]
                 except Exception:
                     raise excpt.InputError(
-                        'The keyword argument '+argument+' was interpreted ' +
-                        'as non-linear momentum compaction factor. ' +
+                        'The keyword argument ' + argument + ' was ' +
+                        'interpreted as non-linear momentum compaction ' +
+                        'factor. ' +
                         'The correct syntax is alpha_n.')
 
         # Setting all the valid non-linear alpha and replacing
         # undeclared orders with zeros
-        self.alpha_orders_defined = [0]
-        for order in range(1, self.alpha_order+1):
+        for order in range(1, alpha_order_max + 1):
             alpha = alpha_n.pop(order, None)
 
             if alpha is None:
                 alpha = 0
                 # This condition can be replaced by 'continue' to avoid
                 # populating the object with 0 programs
             else:
-                self.alpha_orders_defined.append(order)
+                self.alpha_orders.append(order)
 
-            if not isinstance(alpha, dTypes.momentum_compaction):
-                alpha = dTypes.momentum_compaction(alpha, order=order)
+            if not isinstance(alpha, ring_programs.momentum_compaction):
+                alpha = ring_programs.momentum_compaction(alpha, order=order)
             else:
                 if alpha.order != order:
                     raise excpt.InputError(
                         "The order of the datatype passed as keyword " +
                         "argument alpha_%s do not match" % (order))
 
-            self._check_and_set_momentum_compaction(alpha, order)
+            self._check_and_set_alpha_and_orbit(alpha, order)
 
```

Comment:
I would do all the alphas in one go, I think something like this would work:
```python
    allAlphas = {**{'alpha_0': alpha_0, 'alpha_1': alpha_1, 'alpha_2': alpha_2},
                    **{key: kwargs.pop(key) for key in kwargs if 'alpha_' in key}}
    orders = []
    for alpha in allAlphas:
        if allAlphas[alpha] is not None:
            setattr(self, alpha, allAlphas[alpha])
        else:
            setattr(self, alpha, 0)
        orders.append(alpha.split('_')[-1])

    alpha_order_max = np.max(orders)
    for order in range(alpha_order_max):
        if not hasattr(self, 'alpha_'+str(order)):
            setattr(self, 'alpha_'+str(order), 0)
```

Reply:
Will treat it with the other comments from the next PR